pkg/dns/daemon.go

```diff
@@ -82,6 +82,8 @@ func (d *Daemon) internal(ctx context.Context) (bool, error) {
 		return false, nil
 	}
 
+	// TODO: renew certificate
```

Comment:
Yeah, I think renewal should happen here, not in the handler. Any reason you can't move that logic here? Would be fine if its in both places

Reply:
changed my mind. not necessary. the controller is good enough for now i think